polygon/heimdall/entity_store.go

```diff
@@ -0,0 +1,67 @@
+package heimdall
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/json"
+
+	"github.com/ledgerwatch/erigon-lib/kv"
+)
+
+type entityStore interface {
+	GetLastEntityId(ctx context.Context) (uint64, bool, error)
+	PutEntity(ctx context.Context, id uint64, entity any) error
```

Comment:
why `entity any`? you are already creating generic stores - you can make `syncEntity` generic + the stores properly generic

Reply:
In this particular code, didn't need it to be generic, so I've used `any` here.

I didn't want to restrict entityStore to polygon. It is more generic than that and works with any JSON-encodable objects.

In general I prefer to not use generics if possible unless there's a good reason. Generics makes the code more complex and less reusable. They raise a level of cognitive load. Generics tend to proliferate a lot and all the code becomes generic.

With that said, `any` might also be viewed as a form of implicit generics with a downside of not being typesafe. So do you want to ban any usage of `any? How do we decide when to use it and when not? For example, if typecasts or reflection could be avoided, we should use generics instead of any. Are there guidelines about that?

Reply:
"generics make the code less reusable" - sorry but this is fundamentally incorrect statement, the whole reason generics exist is for re-usability

I personally try to avoid using `any` unless I have a really strong reason not to and am forced to use. Getting compile time checks is preferable in my opinion. I don't think in this case we have such a strong reason to use any. And the half the code you have written uses generics while the other uses - any. I find that inconsistent and I think it can be improved.